pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -156,6 +165,35 @@ public void removeSegment(@Nonnull String segmentName) {
     }
   }
 
+  /**
+   * Called when a segment is deleted. The actual handling of segment delete is outside of this method.
+   * This method provides book-keeping around deleted segments.
+   * @param segmentName name of the segment to track.
+   */
+  public void trackDeletedSegment(@Nonnull String segmentName) {
+    // add segment to the cache
+    _deletedSegmentsCache.put(segmentName, true);
+  }
+
+  /**
+   * Check if a segment is recently deleted.
+   *
+   * @param segmentName name of the segment to check.
+   * @return true if segment is in the cache, false otherwise
+   */
+  public boolean isRecentlyDeleted(@Nonnull String segmentName) {
+    return _deletedSegmentsCache.getIfPresent(segmentName) != null;
+  }
+
+  /**
+   * Remove a segment from the deleted cache if it is being added back.
+   *
+   * @param segmentName name of the segment that needs to removed from the cache (if needed)
+   */
+  private void untrackIfDeleted(@Nonnull String segmentName) {
```

Comment:
In my opinion, `untrackDeletedSegment()` seems to be better.

Reply:
I am ambivalent about this. If you feel strongly, I can change it. Will see if Subbu has any input.

Reply:
Or simply, notifySegmentDeleted() and notifySegmentAdded()?

Reply:
I think thats a better a choice. Thanks for the suggestion!